docs/tutorials/otio-file-format-specification.md

```diff
@@ -62,20 +61,40 @@ In order to make the tree structure easy to traverse, we use the name "children"
 
 ## Metadata
 
-TODO: Explain how metadata works and why we do it that way.
+Timeline, Stack, Track, Clip, MediaReferece, and most other OTIO objects all have a `metadata` property.
+This is intended to be a place for you to put information that does not fit into the standard properties.
+The core of OTIO doesn't do anything with metadata, it only carries it along so that adapters, scripts,
+applications, or other workflows can use that metadata however needed. For example, you will find that
+several of the adapters shipped with OTIO use metadata to store information that doesn't (yet) fit into
+the core OTIO schema.
+
+Due to the fact that many different workflows can and will use metadata, it is important to group your
+metadata inside a namespace so that all of those can coexist without stomping on each other. In the example below, you can see that there is metadata on the Timeline and on several Clips for both a hypothetical `my_playback_tool` and `my_production_tracking_system` that could coexist with anything else you add under a similar namespace.
+
+Metadata can also be useful when prototyping new OTIO schemas. An existing object can be extended with metadata which can later be upgraded into a new schema version, or a schemadef plugin.
```

Comment:
Perhaps "migrated into top-level attributes on a new schema version" would be clearer than "upgraded" to convey that the data should move out of the `metadata` dictionary and on to "top-level" properties.